src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -94,10 +100,36 @@ def __init__(
         """
         self._flow_file = flow_file
         self._working_dir = Flow._resolve_working_dir(flow_file, working_dir)
-        self._flow = Flow.from_yaml(flow_file, working_dir=self._working_dir)
-        FlowValidator.ensure_flow_valid_in_batch_mode(self._flow)
+        if Path(flow_file).suffix.lower() in [".yaml", ".yml"]:
+            self._flow = Flow.from_yaml(flow_file, working_dir=self._working_dir)
+            FlowValidator.ensure_flow_valid_in_batch_mode(self._flow)
```

Comment:
Will this PR support the following kind of eager flow?
```yaml
path: ./entry.py
entry: my_flow
```

Reply:
Will support in subsequent pr.